maths/sum_of_digits.py

```diff
@@ -55,9 +55,32 @@ def sum_of_digits_compact(n: int) -> int:
     return sum(int(c) for c in str(abs(n)))
 
 
+def digital_root(n: int) -> int:
+    """
+    finding the digital root of n
+    https://en.wikipedia.org/wiki/Digital_root
+
+    >>> digital_root(394328)
+    2
+    >>> digital_root(-394328)
+    -1
+    >>> digital_root(123)
+    6
+    >>> digital_root(0)
+    0
+    >>> digital_root("109820394")
+    -1
+    """
+
+    try:
+        return n % 9 or n and 9 if n >= 0 else -1
```

Comment:
Should `n and 9` be `n & 9` ?  https://www.geeksforgeeks.org/difference-between-and-and-in-python